### PR TITLE
♻️Refactor amp-ad-network-adsense-impl to move logic related to rive to a separate helper class

### PR DESCRIPTION
**Pull request by riklund**

### CODE REVIEW

**Comment by riklund** [comment]
@lannka @charliereams @keithwrightbos

File: extensions/amp-ad-network-adsense-impl/0.1/test/test-responsive-state.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  createElementWithAttributes,
+} from '../../../../src/dom';
+import {forceExperimentBranch} from '../../../../src/experiments';
+import {layoutRectLtwh} from '../../../../src/layout-rect';
+
+describes.realWin(
+  'responsive-state',
+  {
+    amp: {
+      extensions: ['amp-ad', 'amp-ad-network-adsense-impl'],
```
Comment (lannka) [inline extensions/amp-ad-network-adsense-impl/0.1/test/test-responsive-state.js]: i guess you don't need those extensions in this unit test.

File: extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js [hunk 2]
```diff
@@ -207,39 +156,11 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
...
-          TAG,
-          `Invalid width ${width} for full-width responsive <amp-ad> tag. ` +
-            'Width must be 100vw.'
-        );
-        return false;
-      }
-    }
     return (
-      !!this.element.getAttribute('data-ad-client') && this.isAmpAdElement()
+      (this.responsiveState_ == null ||
```
Comment (lannka) [inline extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js]: one level of `if-else` might make the code a bit easier to read here

File: extensions/amp-ad-network-adsense-impl/0.1/test/test-responsive-state.js (hunk 1 above)
Comment (riklund) [reply extensions/amp-ad-network-adsense-impl/0.1/test/test-responsive-state.js]: done

**Pull request merged**